superset-frontend/src/components/Select/AsyncSelect.tsx

```diff
@@ -425,8 +436,51 @@ const AsyncSelect = forwardRef(
       if (onClear) {
         onClear();
       }
+      fireOnChange();
+    };
+
+    const handleOnBlur = (event: React.FocusEvent<HTMLElement>) => {
+      const tagsMode = !isSingleMode && allowNewOptions;
+      const searchValue = inputValue.trim();
+      // Searched values will be autoselected during onBlur events when in tags mode.
+      // We want to make sure a value is only selected if the user has actually selected it
+      // by pressing Enter or clicking on it.
+      if (
+        tagsMode &&
+        searchValue &&
+        !hasOption(searchValue, selectValue, true)
+      ) {
+        // The search value will be added so we revert to the previous value
+        setSelectValue(selectValue || []);
+      }
+      onBlur?.(event);
     };
 
+    useEffect(() => {
+      if (onChangeCount !== previousChangeCount) {
+        const set = new Set();
+        const array = ensureIsArray(selectValue);
+        array.forEach(item => set.add(getValue(item)));
```

Comment:
You could shorten this.
```suggestion
        const set = new Set(ensureIsArray(selectValue).map(getValue));
```

Reply:
I need the `array` variable bellow. I could do:

```
const array = ensureIsArray(selectValue);
const set = new Set(array.map(getValue));
```